### PR TITLE
[VPlan] Preserve DT (and SCEV) in VPlan-native path

### PR DESCRIPTION
**Pull request by fhahn**
As a follow-up to b2f65e80, use the DTU to also update and preserve
the DT in the native path. This should also allow preserving SCEV in the
native path

### CODE REVIEW

**Comment by tschuett** [comment]
Have you ever considered to build a VPSCEV to decouple from the underlying IR and analysis's? And memory access analysis.

**Review by ayalz — APPROVED** [review]
Thanks for splitting, ship it, with a minor nit. Comment of @tschuett about VPSCEV can be addressed independently.

File: llvm/lib/Transforms/Vectorize/VPlan.cpp [hunk 1]
```diff
@@ -900,9 +900,8 @@ void VPlan::execute(VPTransformState *State) {
   }
 
   State->CFG.DTU.flush();
-  // DT is currently updated for non-native path only.
-  assert(EnableVPlanNativePath || State->CFG.DTU.getDomTree().verify(
-                                      DominatorTree::VerificationLevel::Fast));
+  assert(State->CFG.DTU.getDomTree().verify(
+      DominatorTree::VerificationLevel::Fast));
```
[the remark this answers is not in the document]
Comment (fhahn) [reply llvm/lib/Transforms/Vectorize/VPlan.cpp]: Added, thanks!

**Comment by fhahn** [comment]
> Have you ever considered to build a VPSCEV to decouple from the underlying IR and analysis's? And memory access analysis.

Yep, that's on my TODO list thanks! I already have a motivating use case in mind.

**Pull request merged**